### PR TITLE
Fix for client side hooks, bug fix in template hooks.

### PR DESCRIPTION
**Pull request by bthaeler**
To use hooks on the client side, the globalHookHandler has to be available for addons. Send pass it around using life cycle events in mojito client.

Fixed bug introduced by Caridy in hb.server.js. Hooks start call has to be called before cache code.

### CODE REVIEW

File: lib/app/autoload/mojito-client.client.js [hunk 1]
```diff
@@ -330,7 +330,14 @@ YUI.add('mojito-client', function(Y, NAME) {
...
         init: function(config) {
-            fireLifecycle('pre-init', {config: config});
+            var forwardConfig = {
+                config: config,
+                globalHookHandler: globalHookHandler
+            };
+            fireLifecycle('pre-init', forwardConfig);
+            if (!globalHookHandler) {
+                globalHookHandler = forwardConfig.globalHookHandler;
+            }
```
Comment (drewfish) [inline lib/app/autoload/mojito-client.client.js:346]: Reading the global out of forwardConfig after setting it there is a bit confusing.  I understand what the code is doing but I don't understand _why_.  There should probably be a comment explaining why we need to `globalHookHandler = forwardConfig.globalHookHandler`.

Comment (bthaeler) [inline lib/app/autoload/mojito-client.client.js:346]: Comment added. Also fixed lint issue.

**Comment by drewfish** [comment]
+1.  I ran all tests locally and they all passed.  I'll merge after our internal builds stabilize.

**Comment by drewfish** [comment]
Sorry for the pause, it took a bit longer than I expected :(

**Pull request merged**